core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -160,6 +160,8 @@ abstract class SparkFunSuite
     }
   }
 
+  protected def logForFailedTest(): Unit = {}
```

Comment:
Should we maybe do this only for K8S it tests since the current issue happens to them specifically (for now)?

Reply:
The method `withFixture` which calls this `logForFailedTest` is `final`. 

So either `logForFailedTest` must be here preferably with a default implementation or the `final` should be removed from the `withFixture`  method but in that case the whole method `withFixture` should be copy-pasted to have the log of the failed test in between the "===== TEST OUTPUT " and "===== FINISHED" section (if this is required I think that is better then having it after as it is part of the test).

Reply:
👌